cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -609,12 +614,12 @@ resolveTargets selectPackageTargets selectComponentTarget liftProblem
                            -> [AvailableTarget k]
                            -> Either err [k]
     selectComponentTargets subtarget =
-        either (Left . head) Right
+        either (Left . NonEmpty.head) Right
```

Comment:
I'm leaning towards re-exporing `NonEmpty.head` and `last`  from `Compat.Prelude`

@hvr, opinions?

Reply:
@phadej I'm sympathetic to that :-)

Reply:
Let's do that afterwards.

Reply:
Happy to do so, here or as a follow-up.

Where moving to `NonEmpty`'s head/tail/init/last isn't feasible, should the partial versions be scoped?  Otherwise, you'd have to refer to the imports to see which one is being brought into scope and which is being hidden.

Reply:
then one can

```haskell
-- imports
import Distribution.Compat.Prelude

import qualified Prelude -- add that if you need list head

-- usage

foo = head xs -- NonEmpty variant
bar = Prelude.head ys -- List variant
```